flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/TableSchema.java

```diff
@@ -346,22 +322,137 @@ public static Builder builder() {
 		return new Builder();
 	}
 
+	//~ Tools ------------------------------------------------------------------
+
+	/**
+	 * Tools method to transform arrays of table names and types
+	 * into a {@link TableColumn} list.
+	 */
+	private static List<TableColumn> createTableColumns(
+			String[] fieldNames,
+			TypeInformation<?>[] fieldTypes) {
+			DataType[] fieldDataTypes = fromLegacyInfoToDataType(fieldTypes);
+			validateFields(fieldNames, fieldDataTypes);
+			List<TableColumn> columns = new ArrayList<>();
+		for (int i = 0; i < fieldNames.length; i++) {
+			columns.add(TableColumn.of(fieldNames[i], fieldDataTypes[i]));
+		}
+		return columns;
+	}
+
+	/** Fields sanity check. */
+	private static void validateFields(String[] fieldNames, DataType[] fieldTypes) {
```

Comment:
How about exacting all validation logic into a dedicated `TableSchemaValidator` class? And we should also use `TableColumn` as parameters to validate since this is a more common situation.

Reply:
I would avoid to introduce a class just for some validation tools, different with DescriptorValidator, we only have one `TableSchema`